test/jasmine/tests/plots_test.js

```diff
@@ -54,7 +54,7 @@ describe('Test Plots', function() {
             expect(gd._fullData[1].z).toBe(newData[1].z);
             expect(gd._fullData[1]._empties).toBe(oldFullData[1]._empties);
             expect(gd._fullLayout.scene._scene).toBe(oldFullLayout.scene._scene);
-            expect(gd._fullLayout._plots).toBe(oldFullLayout._plots);
+            expect(gd._fullLayout._plots.plot).toBe(oldFullLayout._plots.plot);
```

Comment:
>  it looks like it reconstructs a new _plots object with the same contents vs. just transferring it?

Correct. Thanks for the headsup. Maybe we could do better.

> Is that what makes the join work correctly? 

I don't think reconstructing `_plots` is necessary to make the joins work correctly.